src/test/java/uk/gov/nationalarchives/pdi/step/jena/serializer/JenaSerializerStepIT.java

```diff
@@ -0,0 +1,192 @@
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+
+package uk.gov.nationalarchives.pdi.step.jena.serializer;
+
+import org.apache.jena.rdf.model.ModelFactory;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.Test;
+import org.mockito.Mockito;
+import org.pentaho.di.core.KettleClientEnvironment;
+import org.pentaho.di.core.exception.KettleException;
+import org.pentaho.di.core.exception.KettleStepException;
+import org.pentaho.di.core.logging.LoggingObjectInterface;
+import org.pentaho.di.core.row.RowMeta;
+import org.pentaho.di.core.row.RowMetaInterface;
+import org.pentaho.di.core.row.value.ValueMetaInteger;
+import org.pentaho.di.trans.step.RowHandler;
+import org.pentaho.di.trans.step.StepDataInterface;
+import org.pentaho.di.trans.steps.mock.StepMockHelper;
+import sun.reflect.generics.reflectiveObjects.NotImplementedException;
+
+import java.io.File;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.LinkedList;
+import java.util.Random;
+
+import static org.junit.jupiter.api.Assertions.assertFalse;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.Mockito.doReturn;
+import static org.mockito.Mockito.when;
+
+public class JenaSerializerStepIT {
+    @BeforeAll
+    public static void setup() throws KettleException {
+        KettleClientEnvironment.init();
+    }
+
+    @Test
+    public void resolves_jena_model_field_variable() throws KettleException {
+        final JenaSerializerStepMeta meta = getMeta();
+        final StepDataInterface data = getData();
+        final StepMockHelper helper = mockHelper();
+        final JenaSerializerStep step = mockStep(helper);
+
+        step.setVariable("jenaModelFieldVar", "model");
+        meta.setJenaModelField("${jenaModelFieldVar}");
+
+        final boolean rowProcessed = step.processRow(meta, data);
+
+        // Would throw if var was not resolved
+        assertTrue(rowProcessed);
+    }
+
+    @Test
+    public void resolves_filename_variable() throws KettleException {
+        // TODO(SL): Use temp folder instead of random file
+        final String expectedFilename = String.valueOf(new Random().nextInt());
+
+        final JenaSerializerStepMeta meta = getMeta();
+        final StepDataInterface data = getData();
+        final StepMockHelper helper = mockHelper();
+        final JenaSerializerStep step = mockStep(helper);
+
+        meta.setJenaModelField("model");
+
+        step.setVariable("filenameVar", expectedFilename);
+        meta.setFileDetail(
+                new JenaSerializerStepMeta.FileDetail() {{
+                    filename = "${filenameVar}";
+                }});
+
+        // First pass create model, second serialise
+        step.processRow(meta, data);
+        step.processRow(meta, data);
+
+        File f = new File(expectedFilename);

```

Comment:
Within our code please use the newer `Path` API instead of `File`.
As many legacy APIs use File, you will find methods `File#toPath()` and `Path#toFile()` useful.